src/core/core_utils.js

```diff
@@ -340,6 +340,7 @@ function _collectJS(entry, xref, list, parents) {
         code = js;
       }
       code = code && stringToPDFString(code);
+      code = code.replace(/\u0000+/, "");
```

Comment:
Given that `code` can still be undefined here, this could potentially throw.
Also, don't you need to do a *global* replacement; hence I'd expect that you actually want the following here:
```suggestion
      code = code && stringToPDFString(code).replace(/\u0000/g, "");
```